### DISCUSSION TITLE
Inconsistent MDC properties in health checks

### DISCUSSION THREAD

**Discussion by atanasnik** [quarkusio/quarkus#41711] [Q&A] [answered]
Hi,

I am asking for help with a problem that I have recently noticed with health checks.

As my team has been using Quarkus `3.6.1` in our project, we were used to adding an MDC key-value property to each one of our health checks (using `SmallRyeHealthReporter`) which indicates the name of the current health check operation. Health checks were logging on a single thread and we had no issues with MDC and logging. 

We recently decided to add [context propagation](https://quarkus.io/guides/context-propagation) and update Quarkus to `3.12.0`. Now, it seems that each of the health checks is logging on a different thread, and apparently, a key-value property that has been added to MDC in a health check might appear in another health check's log, which leaves us with very inconsistent logs due to the threads' behavior. The distribution of MDC properties across health check logs seems non-deterministic. 

I suppose this problem comes from the updated Quarkus version as I can only reproduce it after updating it and I am not sure if health checks are meant to be run this way on multiple threads and if there is a proper way to fix the issue we are experiencing.

Any advice is appreciated.

[a comment by an automated account is not shown]

**Comment by xstefank** [comment]
Reproduced and acknowledged as a bug - https://github.com/quarkusio/quarkus/issues/41746.

**Comment by xstefank** [comment, marked as answer]
@atanasnik this is now fixed in 3.12.3. Can you verify please?

**Reply by atanasnik** [reply]
Each of the health checks is logging on a different thread, but it seems context is passed in a correct way, so I can confirm that this update now solves the problem we had. Thank you!
